### ISSUE TITLE
mesh_get_index returns a value of -1 when a particle lies on mesh minimum boundary

### ISSUE THREAD

**Issue by goodman17c** [CEMeNT-PSAAP/MCDC#108]
The mesh_get_index returns a value of -1 when a particle is at the lower boundary of a mesh in any of space or time. This is problematic for having an initial condition or boundary source. Often, the effect is unnoticed as Python will treat the -1 index to reference the last element of the array without error or warning.

A workaround has been to modify the mesh in the input file with some small perturbation.

**Comment by ilhamv** [comment]
That is true. The current solution is that users need to move the boundary/point source into the domain by manually specifying it in the `mcdc.source`.

**Comment by ilhamv** [comment]
This should be resolved per the new geometry tolerance treatment #244 .
